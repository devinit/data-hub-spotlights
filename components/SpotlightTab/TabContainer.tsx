import React, { Children, FunctionComponent, isValidElement } from 'react';
import { TabContent } from './TabContent';

interface TabContainerProps {
  id: string;
  label: string;
  active?: boolean;
}

const TabContainer: FunctionComponent<TabContainerProps> = ({ id, active, label, children }) => {
  const renderContent = () => {
    return Children.map(
      children,
      (child) => isValidElement(child) && child.type === TabContent ? child : null
    );
  };

  return (
    <section className="tabs__container" id={ id }>
      <input className="tabs__input" type="radio" name="sections" id={ `${id}-option` } checked={ active }/>
      <label className="tabs__label" htmlFor={ `${id}-option` }>{ label }</label>
      { renderContent() }
    </section>
  );
};

export { TabContainer };
