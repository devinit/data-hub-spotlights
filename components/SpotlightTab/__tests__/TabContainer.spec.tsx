/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { TabContainer } from '../TabContainer';

describe('TabContainer', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<TabContainer id="container" label="My Label" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
