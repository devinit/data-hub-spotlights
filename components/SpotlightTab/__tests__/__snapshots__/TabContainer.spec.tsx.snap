// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabContainer renders correctly 1`] = `
<section
  className="tabs__container"
  id="container"
>
  <input
    className="tabs__input"
    id="container-option"
    name="sections"
    type="radio"
  />
  <label
    className="tabs__label"
    htmlFor="container-option"
  >
    My Label
  </label>
</section>
`;
